subsys/net/lib/sockets/sockets.c

```diff
@@ -173,6 +175,46 @@ int z_impl_zsock_socket(int family, int type, int proto)
 	return -1;
 }
 
+int z_impl_zsock_socketpair(int family, int type, int proto, int sv[2]) {
+	int res;
+
+	if (family != AF_UNIX) {
+		errno = EAFNOSUPPORT;
+		res = -1;
+		goto out;
+	}
+
+	if (!(type == SOCK_STREAM || type == SOCK_DGRAM || type == SOCK_RAW)) {
+		errno = EPROTOTYPE;
+		res = -1;
+		goto out;
+	}
+
+	if (proto != 0) {
+		errno = EPROTONOSUPPORT;
+		res = -1;
+		goto out;
+	}
+
+	if (sv == NULL) {
+		/* not listed in the normative standard, but probably safe */
+		errno = EINVAL;
+		res = -1;
+		goto out;
+	}
+
+	/* TODO: implement pipe(2) in the posix subsystem */
+	/* TODO: call z_impl_zposix_pipe(sv) ?*/
+	r = -1;
+	errno = ENOSYS;
+	goto out;
+
+	res = 0;
+
+out:
+	return res;
+}
+
 #ifdef CONFIG_USERSPACE
```

Comment:
I think it is missing the verification handler for this syscall. If that's correct your test is probably no calling this syscall from usermode